extensions/rich_text_components/Image/protractor.js

```diff
@@ -17,19 +17,180 @@
+    'troke" id="group0"/></g><g transform="matrix(1 0 0 1 0 0)" id="group0">' +
+    '<polygon style="stroke: rgb(0,255,0); stroke-width: 1; stroke-dasharray' +
+    ': none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: mi' +
+    'ter; stroke-miterlimit: 4; fill: rgb(0,255,0); fill-rule: nonzero; opac' +
+    'ity: 1; vector-effect: non-scaling-stroke" points="0,0 0,30 0,-30 0,0 "' +
+    ' id="group0"/></g></g></g></g></g><g transform="matrix(1 0 0 1 50 50)">' +
+    '<g style=""><g transform="matrix(1 0 0 1 0 0)" id="group1"><circle styl' +
+    'e="stroke: rgb(0,128,0); stroke-width: 4; stroke-dasharray: none; strok' +
+    'e-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-m' +
+    'iterlimit: 4; fill: rgb(255,255,0); fill-rule: nonzero; opacity: 1;" cx' +
+    '="0" cy="0" r="40" id="group1"/></g></g></g></svg>')
 };
 
-var expectComponentDetailsToMatch = async function(elem, name) {
-  // The original filepath is not recorded and so cannot be checked.
-  expect(await elem.getAttribute('filepath-with-value')).toMatch(name);
+var customizeComponent = async function(modal, selectedAction, args, altText) {
+  // The arguments for this function are as follows:
+  //   - modal: The selector for the customisation arguments modal.
+  //   - selectedAction: Represents the user action. Image RTE component
+  //       allows uploading an image from the system or creation of an
+  //       image from scratch using the diagram editor. This argument
+  //       takes the value 'create' or 'upload'.
+  //   - args: The input to the editor. If the selected action is 'create',
+  //       args would be a list of shapes and for 'upload' action, it
+  //       would represent a filepath.
+  //   - altText: The alt text for the image.
+  if (selectedAction === 'create') {
+    await action.click(
+      'Create button',
+      modal.element(by.css('.protractor-test-create-image')));
+    for (var i = 0; i < args.length; i++) {
+      var shapeClass = '.protractor-test-create-' + args[i];
+      var shapeTool = modal.element(by.css(shapeClass));
+      await waitFor.elementToBeClickable(
+        shapeTool,
+        'Could not click on the required tool');
+      var customTools = ['bezier', 'piechart', 'svgupload'];
+      if (customTools.includes(args[i])) {
+        await shapeTool.click();
+        if (args[i] === 'svgupload') {
+          var imageUploadInput = element(
+            by.css('.protractor-test-photo-upload-input'));
+          absPath = path.resolve(__dirname, './circle.svg');
+          await waitFor.presenceOf(
+            imageUploadInput,
+            'Image upload input element took too long to load');
+          await imageUploadInput.sendKeys(absPath);
+        }
+      }
+      await shapeTool.click();
+    }
+    var altTextInputElement = (
+      element(by.css(
+        '[placeholder = "Description of Image (Example : George Handel, ' +
+        '18th century baroque composer)"]')));
```

Comment:
Can we use a `protractor-test-*` HTML class here instead? I know this was carried over from the old code though, so I'm okay with leaving it if it's going to be a lot of work to change

Reply:
I think the reason we do this is because it is hard to specify a specific selector for this element, it is too generic and gets used in several places (and sometimes there can be several elements with the same selector on the screen). The "alt text" input field will render a "schema-based-unicode-editor" and this will be common for all editors that take a unicode input. Considering this, I think it should be ok to leave this as-is for now.